src/Common/src/Interop/Unix/System.Native/Interop.GetUnixArchitecture.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [DllImport(Libraries.SystemNative, SetLastError = true)]
+        internal static extern int GetUnixArchitecture();
+
+        internal enum ProcessorArchitecture
+        {
+            x86,
+            x64,
+            ARM
```

Comment:
Other existing runtimes (mono, Unity) do support arm64 already. You should add it to the public contract at least so that this contract can be implemented by other runtimes.

And arm64 is in progress for CoreCLR, so adding the couple of lines for it in the implementation would be nice too.